test/mfa.test.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-undef */
+import * as Factory from './lib/factory.js';
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+describe('Multi-Factor Authentication (MFA)', function () {
+  beforeEach(async function () {
+    localStorage.clear();
+    this.application = await Factory.createInitAppWithRandNamespace();
+    this.email = Uuid.GenerateUuidSynchronously();
+    this.password = Uuid.GenerateUuidSynchronously();
+  });
+
+  afterEach(function () {
+    this.application.deinit();
+    this.application = null;
+    localStorage.clear();
+  });
+
+  it('mfa flow', async function () {
```

Comment:
This test is too long. One test is also not enough for a feature like this. Should be broken up into different tests. One test should be "removing 2FA should allow sign in without token" for example.

Reply:
Should have separate tests for enabling and disabling 2FA as well, that just ensures you are prompted or not prompted.